integration_tests/tests/integration/in_cluster.rs

```diff
@@ -333,3 +333,244 @@ async fn in_cluster_fixed_size() {
     )
     .await;
 }
+
+mod rate_limits {
+    #![cfg(feature = "in-cluster-rate-limits")]
+
+    use super::InClusterJanusPair;
+    use http::Method;
+    use janus_aggregator_core::task::QueryType;
+    use janus_core::{test_util::install_test_trace_subscriber, vdaf::VdafInstance};
+    use janus_messages::{AggregationJobId, CollectionJobId, TaskId};
+    use rand::random;
+    use reqwest::StatusCode;
+    use serde::Deserialize;
+    use std::{env, fs::File};
+    use url::Url;
+
+    /// Configuration for the rate limit test. We need to know the QPS and the window over which
+    /// it is enforced so that we can send the appropriate number of requests. We load this config
+    /// from a file so that an integration test harness that knows what rate limits have been
+    /// deployed to the cluster we test against can set the values.
+    #[derive(Deserialize)]
+    struct TestConfig {
+        window: u64,
+        upload_qps: u64,
+        aggregation_job_qps: u64,
+        collection_job_qps: u64,
+    }
+
+    impl TestConfig {
+        fn load() -> Self {
+            serde_json::from_reader(
+                File::open(
+                    env::var("JANUS_E2E_RATE_LIMIT_TEST_CONFIG")
+                        .unwrap()
+                        .as_str(),
+                )
+                .unwrap(),
+            )
+            .unwrap()
+        }
+    }
+
+    struct RequestUrlParameters<'a> {
+        leader_url: &'a Url,
+        helper_url: &'a Url,
+        first_task_id: &'a TaskId,
+        second_task_id: &'a TaskId,
+    }
+
+    async fn run_rate_limit_test(
+        request_url_maker: &dyn Fn(&RequestUrlParameters) -> (Url, Url),
+        rate_limit_picker: &dyn Fn(&TestConfig) -> u64,
+        method: Method,
+    ) {
+        install_test_trace_subscriber();
+        let test_config = TestConfig::load();
```

Comment:
nit: could we cache the result of loading the test config? Since this is a test & errors lead to a panic, we could use a static `Once` in the `TestConfig::load` method to load the test config, with the `Once` roughly wrapping the code that currently exists in `TestConfig::load`.

(I'm slightly curious about why we need to read this information from a file -- I suppose we can't hardcode it in the tests because ultimately the rate-limit config is determined outside of this code. Is that about correct?)

Reply:
Check out https://github.com/divviup/janus-ops/pull/1205 to see how Terraform will drop the JSON test config file that these tests then load. Since the rate limits are actually governed by the .tfvars files in janus-ops, I don't want to duplicate those values here, because then we'd have to make two changes across two projects every time we tweak the rate limits in Kind deployments.

This test is defined here in Janus, but as I note in the PR description, this isn't actually about Janus but rather exercising the Caddy rate limiting reverse proxy we set up over in janus-ops. It's just that this crate, since it already knows how to deal with a pair of Janus aggregators in a Kind cluster, is a convenient place to do that. I think maybe we could kick all of the `integration_tests::in_cluster` module out of the Janus repository, but I'm not sure I want to deal with that now: we already had a bunch of Kubernetes-specific goop here that isn't strictly essential to running or testing Janus, and instead reflects how Divvi Up happens to manage it.

Reply:
Agreed that we can wait to move integration tests out of the `janus` repo -- arguably we should have put them in `janus_ops` (or their own repository) to start, but at this point we have integration tests already defined in `janus`, so this is the right place to put this code unless/until we decide to move all integration tests.